datacube/index/abstract.py

```diff
@@ -923,6 +923,12 @@ def find_less_mature(self, ds: Dataset, delta: int = 500) -> Iterable[Dataset]:
         less_mature = []
         assert delta >= 0
 
+        # in case value winds up getting passed in as a bool
```

Comment:
Can we please type-hint delta as Union[int, bool] then? 

Note in docstring that int is preferred.